### PR TITLE
refactor: TrainingPipeline interface

### PR DESCRIPTION
**Pull request by zzsi**

### CODE REVIEW

**Comment by zzsi** [comment]
[worker:drifter] Implementation complete. PR #218 adds the TIDE cross-architecture distillation example at `examples/generative/llm/tide_dllm_distill/`.

**What's implemented:**
- Self-contained `train.py` implementing TIDE-Shared distillation (Pipeline B):
  - **TIDAL**: cosine-interpolated soft targets with dual-axis scheduling
  - **CompDemo**: complementary mask-splitting for better teacher signals at high masking ratios
  - Configurable teacher/student models, dataset, and all hyperparameters
- `predict.py` for inference with distilled models (supports pre-trained TIDE checkpoints from HuggingFace)
- Full CVlization conventions: Dockerfile, build/train/predict/test presets, example.yaml

**Verified on drifter:**
- Docker build: ✅
- Training smoke test (self-distillation, 10 steps): ✅ loss 5.37→4.59, TIDAL lambda 0.38→0.88
- Inference smoke test: ✅ "What is 2+2?" → "4"

**Usage:**
```bash
cvl run tide-dllm-distill build
cvl run tide-dllm-distill test          # smoke test
cvl run tide-dllm-distill train -- --teacher-model WeDLM/WeDLM-8B-Instruct  # full distillation
cvl run tide-dllm-distill predict -- --model tide-wedlm-shared  # use pre-trained TIDE model
```

**Comment by zzsi** [comment]
[worker:drifter] Ran `verify-training-pipeline` skill. All checks pass.

## Verification Checklist Results

| # | Check | Status | Details |
|---|-------|--------|---------|
| 1 | **Structure** | ✅ PASS | All required files present: example.yaml, Dockerfile, build.sh, train.sh, train.py, README.md, requirements.txt, .gitignore |
| 2 | **Build** | ✅ PASS | `./build.sh` completes, image 9.82 GB |
| 3 | **Training starts** | ✅ PASS | Container starts, dataset loads (alpaca), model initializes, training loop runs |
| 4 | **Metrics improve** | ✅ PASS | Loss decreases 5.37 → 4.59 (10 steps self-distillation). TIDAL lambda correctly ramps 0.38 → 0.88 via cosine schedule |
| 5 | **Central caching** | ✅ PASS | Model cached at `~/.cache/huggingface/hub/models--dllm-hub--Qwen3-0.6B-diffusion-bd3lm-v0.1/`. No downloads into repo |
| 6 | **Lazy downloading** | ✅ PASS | Second run reuses cached model and dataset |
| 7 | **Outputs / .gitignore** | ✅ PASS | `.gitignore` covers `checkpoint-*/`, `*.safetensors`, `result.txt`, `result.json` etc. Cleaned root-owned artifacts via Docker |
| 8 | **CUDA_VISIBLE_DEVICES** | ✅ PASS | Both train.sh and predict.sh forward `CUDA_VISIBLE_DEVICES` |
| 9 | **Documentation** | ✅ PASS | README covers all usage modes |
| 10 | **Verification metadata** | ✅ PASS | `example.yaml` updated with `last_verified: 2026-05-22` |

## Fixes Applied

- Added `CUDA_VISIBLE_DEVICES` forwarding to predict.sh (was missing)
- Added `checkpoint-*/`, `result.txt`, `result.json` to `.gitignore` (checkpoints land in workspace root, not `checkpoints/`)
- Added `verification` field to example.yaml

## GPU/VRAM Details

- **GPU**: NVIDIA RTX PRO 6000 Blackwell (98 GB VRAM)
- **Peak VRAM (self-distillation)**: 8.5 GB (8.7% of total)
- **Training time (10 steps)**: 2.0s
- **Inference**: "What is 2+2?" → "4" ✅

## Verification Skill Used

`verify-training-pipeline` (`.claude/skills/verify-training-pipeline/SKILL.md`)
